red_queen/games/applications/run_bv.py

```diff
@@ -13,14 +12,35 @@
 
 from red_queen.games.applications import backends, run_qiskit_circuit
 
+import random
+
+import numpy
 
 QASM_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "qasm")
-SECRET_STRING = "110011"
 
 
+@pytest.fixture
+def get_num_qubits(request):
+    return request.config.getoption("--num_qubits")
+
+
+def generate_num_qubits(get_num_qubits):
+    user_params = requests.get(get_num_qubits)
+    user_params = user_params.split(":")
+    if user_params[0] == "linear":
+        sweep_list = numpy.linspace(int(user_params[1]), int(user_params[2]), int(user_params[3]))
+        sweep_list = [int(i) for i in sweep_list]
+    if user_params[0] == "log":
+        sweep_list = numpy.logspace(int(user_params[1]), int(user_params[2]), int(user_params[3]))
+        sweep_list = [int(i) for i in sweep_list]
+    for i in sweep_list:
+        num_qubits = i
+        SECRET_STRING = bin(random.getrandbits(i - 1))[2:]
+
+
+@pytest.mark.parametrize("num_qubits", sweep_list)
```

Comment:
I don't think using the decorator here to parameterize this function will be sufficient to update the benchmark to run with `n` qubits.  I think you'll need to add the decorator to the `bench_qiskit_bv`  function and take `num_qubits` as a parameter on the function definition. Then from there you'll need to update the function's code to call `build_bv_circuit` with the secret string from the parameter and pass that returned circuit directly to `run_qiskit_circuit`. Right now the benchmark function is hard coded to load the circuit from the in repo qasm file.

Reply:
I tried doing this, is the way I implemented it correct?